src/core_codemods/with_threading_lock.py

```diff
@@ -0,0 +1,51 @@
+import libcst as cst
+from codemodder.codemods.base_codemod import ReviewGuidance
+from codemodder.codemods.api import SemgrepCodemod
+
+
+class WithThreadingLock(SemgrepCodemod):
+    NAME = "bad-lock-with-statement"
+    SUMMARY = "Replace deprecated usage of threading.Lock context manager"
+    REVIEW_GUIDANCE = ReviewGuidance.MERGE_AFTER_CURSORY_REVIEW
+    DESCRIPTION = "Separates threading lock instantiation and call with `with` statement into two steps."
+
+    @classmethod
+    def rule(cls):
+        return """
+        rules:
+          - patterns:
+            - pattern: |
+                with $BODY:
+                    ...
+            - metavariable-pattern:
+                metavariable: $BODY
+                patterns:
+                - pattern: threading.Lock()
+            - pattern-inside: |
+                import threading
+                ...
+            - focus-metavariable: $BODY
+        """
+
+    def leave_With(self, original_node: cst.With, updated_node: cst.With):
+        if original_node.items and self.node_is_selected(original_node.items[0]):
```

Comment:
You can have several items within a `with` (i.e `with a as b, c as d, ...`, are we purposefully avoiding those cases?

Reply:
Yes I think we can avoid those for now. It seems like it will be a bit harder to match those using semgrep, but I think the current implementation probably covers 90% of cases anyway.